client/src/main/java/io/split/service/SplitHttpClient.java

```diff
@@ -0,0 +1,31 @@
+package io.split.service;
+
+import io.split.engine.common.FetchOptions;
+import io.split.telemetry.domain.enums.HttpParamsWrapper;
+import io.split.client.dtos.SplitHttpResponse;
+
+import org.apache.hc.core5.http.HttpEntity;
+import java.io.IOException;
+import java.net.URI;
+import java.util.Map;
+
+public interface SplitHttpClient {
+    /**
+     * Wrapper for HTTP get method
+     * @param uri the URL to be used
+     * @param options The FetchOptions object that contains headers.
+     * @return The response structure SplitHttpResponse
+     */
+    public SplitHttpResponse get(URI uri, FetchOptions options);
```

Comment:
should we allow custom headers here as well?

Reply:
makes sense, will do